src/main/java/com/minecolonies/coremod/client/gui/questlog/WindowQuestLogBase.java

```diff
@@ -0,0 +1,208 @@
+package com.minecolonies.coremod.client.gui.questlog;
+
+import com.ldtteam.blockui.Pane;
+import com.ldtteam.blockui.PaneBuilders;
+import com.ldtteam.blockui.controls.Button;
+import com.ldtteam.blockui.controls.ButtonImage;
+import com.ldtteam.blockui.controls.Text;
+import com.ldtteam.blockui.views.ScrollingList;
+import com.minecolonies.api.colony.ICitizenDataView;
+import com.minecolonies.api.colony.IColonyView;
+import com.minecolonies.api.quests.IQuestInstance;
+import com.minecolonies.coremod.client.gui.AbstractWindowSkeleton;
+import net.minecraft.network.chat.Component;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.List;
+import java.util.function.Function;
+
+import static com.minecolonies.api.util.constant.WindowConstants.BUTTON_NEXTPAGE;
+import static com.minecolonies.api.util.constant.WindowConstants.BUTTON_PREVPAGE;
+import static com.minecolonies.api.util.constant.translation.GuiTranslationConstants.QUEST_LOG_TRACK_CITIZEN_TEXT;
+import static com.minecolonies.coremod.client.gui.questlog.Constants.BUTTON_QUEST_LOCATOR;
+
+/**
+ * Quest log window base tab.
+ */
+public abstract class WindowQuestLogBase<T> extends AbstractWindowSkeleton
+{
+    /**
+     * The colony this quest log is attached to.
+     */
+    @NotNull
+    private final IColonyView colonyView;
+
+    /**
+     * ScrollList with all the quests.
+     */
+    private ScrollingList questsList;
+
+    /**
+     * The list of quest items
+     */
+    private List<T> questItems;
+
+    /**
+     * Default constructor.
+     */
+    protected WindowQuestLogBase(final String resource, final @NotNull IColonyView colonyView)
+    {
+        super(resource);
+        this.colonyView = colonyView;
+        registerButton(BUTTON_PREVPAGE, this::openTabPage);
+        registerButton(BUTTON_NEXTPAGE, this::openTabPage);
+        registerButton(BUTTON_QUEST_LOCATOR, this::locateCitizenClickedInternal);
+    }
+
+    /**
+     * Open the previous or next tab.
+     *
+     * @param button the button which was clicked.
+     */
+    private void openTabPage(Button button)
+    {
+        Function<IColonyView, AbstractWindowSkeleton> nextWindow = button.getID().equals("prevPage") ? getNavigationPrev() : getNavigationNext();
```

Comment:
you do that using switch component, and AbstractWindowSkeleton has builtin detection for multi page uis